File: frontend/src/app/service/comment.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { firstValueFrom } from 'rxjs';
import { CreateCommentRequest } from '../contracts/requests';
import { Comment } from '../contracts/models';

@Injectable({
  providedIn: 'root',
})
export class CommentService {
  constructor(private httpClient: HttpClient) {}

  public async createComment(
    userId: number,
    trailId: number,
    request: CreateCommentRequest
  ): Promise<Comment> {
    return await firstValueFrom(
      this.httpClient.post<Comment>(
        `http://localhost:7070/comment/${userId}/${trailId}`,
        request
      )
    );
  }

  public async updateComment(
    userId: number,
    commentId: number,
    request: CreateCommentRequest
  ): Promise<Comment> {
    return await firstValueFrom(
      this.httpClient.put<Comment>(
        `http://localhost:7070/comment/${userId}/${commentId}`,
        request
      )
    );
  }

  public async deleteComment(
    userId: number,
    commentId: number
  ): Promise<Comment> {
    return await firstValueFrom(
      this.httpClient.delete<Comment>(
        `http://localhost:7070/comment/${userId}/${commentId}`
      )
    );
  }
}

File: backend/src/main/java/me/matej/hikingtrailmapper/service/CommentService.java
package me.matej.hikingtrailmapper.service;

import me.matej.hikingtrailmapper.contracts.CreateCommentRequest;
import me.matej.hikingtrailmapper.dtos.CommentDto;

public interface CommentService {
    CommentDto createComment(Long userId, Long trailId, CreateCommentRequest request);
    CommentDto updateComment(Long userId, Long commentId, CreateCommentRequest request);
    CommentDto deleteComment(Long userId, Long commentId);
}
